test/utils/createAlgoliaResponse.ts

```diff
@@ -0,0 +1,38 @@
+import { Response } from 'algoliasearch';
+
+/**
+ * Create a response conforming a real Algolia response from a subset of it.
+ * Only parameters which always return are given a default value.
+ *
+ * If not given, nbPages is computed from nbHits & hitsPerPage
+ */
+export default function createAlgoliaResponse<THit = any>(
+  subset: Partial<Response<THit>> = {}
+): Response<THit> {
+  const {
+    page = 0,
+    hitsPerPage = 20,
+    nbHits = 0,
+    nbPages,
+    processingTimeMS = 0,
+    hits = [],
+    query = '',
+    params = '',
+    exhaustiveNbHits = true,
+    exhaustiveFacetsCount = true,
```

Comment:
TS fails here, but that's fixed by https://github.com/DefinitelyTyped/DefinitelyTyped/pull/36859